packages/relic/lib/src/headers/accept_header.dart

```diff
@@ -0,0 +1,49 @@
+part of '../headers.dart';
+
+/// A class representing the HTTP Accept header.
+///
+/// This class manages MIME types and optional quality values (`q` values).
+/// It supports parsing and generating the appropriate header string.
+class AcceptHeader {
+  /// The list of accepted MIME types.
+  final List<MediaType> mediaTypes;
+
+  /// Constructs an [AcceptHeader] instance with the specified media types.
+  AcceptHeader(this.mediaTypes);
+
+  /// Parses the Accept header value and returns an [AcceptHeader] instance.
+  ///
+  /// This method splits the header by commas, trims each media type, and processes
+  /// the MIME types along with optional `q` values.
+  factory AcceptHeader.fromHeaderValue(List<String> value) {
+    final mediaTypes = value
+        .fold(
+          [],
+          (a, b) => [
+            ...a,
+            ...b.split(',').map((directive) => directive.trim()),
+          ],
+        )
+        .where((e) => e.isNotEmpty)
+        .map((part) => MediaType.parse(part.trim()))
+        .toList();
+
+    return AcceptHeader(mediaTypes);
+  }
+
+  /// Static method that attempts to parse the Accept header and returns `null` if the value is `null`.
+  ///
+  /// This method safely parses the Accept header value or returns `null` if the input is invalid or `null`.
+  static AcceptHeader? tryParse(List<String>? value) {
+    if (value == null) return null;
+    return AcceptHeader.fromHeaderValue(value);
```

Comment:
@vlidholt The idea is that if the value is null, we should return null. However, if it's not null and has some value, we should throw an error because the value is invalid. If they provide a value but see the header as null, they won’t understand what's going on.
What do you suggest i do here ?

Reply:
Please see my overarching comments. A `tryParse` should never throw an exception. Having the null check inside the parse method adds little value. The parsers shouldn't take nullable values, instead do the null check when calling the method. E.g., with `?.`.